kie-server-parent/kie-server-services/kie-server-services-dmn/src/main/java/org/kie/server/services/dmn/ModelEvaluatorServiceBase.java

```diff
@@ -204,8 +217,128 @@ public static DMNItemDefinitionInfo itemDefinitionToInfo(ItemDefinition itemDef)
         }
     }
 
+    public Response evaluateModel(String containerId, String modelId, String contextPayload, boolean asDmnResult, String decisionServiceId) {
+        try {
+            KieContainerInstanceImpl kContainer = context.getContainer(containerId, ContainerLocatorProvider.get().getLocator());
+            DMNRuntime dmnRuntime = KieRuntimeFactory.of(kContainer.getKieContainer().getKieBase()).get(DMNRuntime.class);
+
+            List<DMNModel> modelsWithID = dmnRuntime.getModels().stream().filter(m -> m.getName().equals(modelId)).collect(Collectors.toList());
+            if (modelsWithID.isEmpty()) {
+                return Response.serverError().entity("No model identifies with modelId: " + modelId).build();
```

Comment:
Same here, I think that HTTP 404 would be proper response.

Reply:
as above, please let me know (above)